### PR TITLE
[MI 2837]: Done the review fixes of a gitlab PR #301

### PR DESCRIPTION
**Pull request by Nityanand13**
[PR: #301](https://github.com/mattermost/mattermost-plugin-gitlab/pull/301)

### CODE REVIEW

File: webapp/package-lock.json
Comment (SaurabhSharma-884) [inline webapp/package-lock.json:1]: These changes should not be here

File: webapp/src/components/link_tooltip/link_tooltip.jsx [hunk 1]
```diff
@@ -24,6 +24,11 @@ const LINK_TYPES = {
     ISSUES: 'issues',
 };
 
+export function getInfoAboutLink(href,hostname) {
+    const [owner, repo, , type, number] = href.split(`${hostname}/`)[1].split('/');
+    return [owner, repo, type, number];
+}
+
```
Comment (SaurabhSharma-884) [inline webapp/src/components/link_tooltip/link_tooltip.jsx:31]:
```suggestion
export const  getInfoAboutLink = (href, hostname) => {
    return href.split(`${hostname}/`)[1].split('/');
}

```

Comment (SaurabhSharma-884) [reply webapp/src/components/link_tooltip/link_tooltip.jsx:31]: Check if this works

File: webapp/src/components/link_tooltip/link_tooltip.jsx [hunk 2]
```diff
@@ -94,94 +98,95 @@ export const LinkTooltip = ({href, connected, gitlabURL}) => {
...
-                            title={data.repo}
-                            href={href}
-                        >
-                            {data.repo}
-                        </a>
-                        {' on '}
-                        <span>{date}</span>
-                    </div>
+    if (!data) {
+        return null
```
Comment (SaurabhSharma-884) [inline webapp/src/components/link_tooltip/link_tooltip.jsx:102]:
```suggestion
        return null;
```

File: webapp/src/components/link_tooltip/link_tooltip.jsx [hunk 3]
```diff
@@ -94,94 +98,95 @@ export const LinkTooltip = ({href, connected, gitlabURL}) => {
...
                             </div>
+                        )}
+
+                        <div className='see-more mt-1'>
+                            <a
+                                href={href}
+                                target='_blank'
+                                rel='noopener noreferrer'
+                            >
+                                See more
```
Comment (SaurabhSharma-884) [inline webapp/src/components/link_tooltip/link_tooltip.jsx:166]:
```suggestion
                                {'See more'}
```

Comment (SaurabhSharma-884) [reply webapp/src/components/link_tooltip/link_tooltip.jsx:166]: Please check eslint configuration

File: webapp/src/components/link_tooltip/link_tooltip.test.jsx [hunk 4]
```diff
@@ -0,0 +1,44 @@
+import {getInfoAboutLink} from './link_tooltip'
+
+describe("getInfoAboutLink", () => {
+
+    it("should return the correct owner, repo, type, and number when given a valid GitLab href and hostname with all required information", () => {
```
Comment (SaurabhSharma-884) [inline webapp/src/components/link_tooltip/link_tooltip.test.jsx:5]:
```suggestion
    it("Should return the correct owner, repo, type, and number when given a valid GitLab href and hostname with all required information", () => {
```

Comment (SaurabhSharma-884) [reply webapp/src/components/link_tooltip/link_tooltip.test.jsx:5]: Check everywhere

File: webapp/src/utils/regex_utils.test.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import {validateGitlabURL} from './regex_utils';
+
+describe("validateGitlabURL", () => {
```
Comment (SaurabhSharma-884) [inline webapp/src/utils/regex_utils.test.js:3]: Can add a descriptive statement here instead of `validateGitlabURL`

File: webapp/package-lock.json
Comment (Nityanand13) [reply webapp/package-lock.json:1]: @SaurabhSharma-884 Is there any specific reason behind that?

Comment (SaurabhSharma-884) [reply webapp/package-lock.json:1]: This is auto-generated when we run `npm install` also it may give rise to lots of conflicts while merging or rebasing.

**Review by SaurabhSharma-884 — APPROVED** [review]

File: webapp/package-lock.json
Comment (Nityanand13) [reply webapp/package-lock.json:1]: @SaurabhSharma-884 There are a few new packages that I installed due to which it is auto-generated.

File: webapp/src/components/link_tooltip/link_tooltip.jsx [hunk 6]
```diff
@@ -94,94 +97,95 @@ export const LinkTooltip = ({href, connected, gitlabURL}) => {
...
                                     <span
-                                        title={data.source_branch}
-                                        className='commit-ref'
+                                        key={index}
+                                        className='label mr-1'
+                                        title={label.description}
+                                        style={{backgroundColor: label.color}}
                                     >
-                                        {data.source_branch}
+                                        <span style={{color: label.text_color}}>{label.name}</span>
```
Comment (sooraj-shukla) [inline webapp/src/components/link_tooltip/link_tooltip.jsx:179]: Avoid inline styles.

File: webapp/src/components/link_tooltip/link_tooltip.jsx [hunk 7]
```diff
@@ -94,94 +97,95 @@ export const LinkTooltip = ({href, connected, gitlabURL}) => {
...
-                                    </span>
-                                    <span className='mx-1'>←</span>
+                        {/* Labels */}
+                        <div className='labels mt-3'>
+                            {data.labels && data.labels_with_details?.length && data.labels_with_details.map((label, index) => {
+                                return (
                                     <span
-                                        title={data.source_branch}
-                                        className='commit-ref'
+                                        key={index}
```
Comment (sooraj-shukla) [inline webapp/src/components/link_tooltip/link_tooltip.jsx:174]: Avoid using `index` as key.

File: webapp/package.json [hunk 8]
```diff
@@ -7,38 +7,58 @@
...
+        "transform": {
+            "^.+\\.[t|j]sx?$": "babel-jest"
+        },
+        "moduleNameMapper": {
+            "^.+\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$": "identity-obj-proxy",
+            "^.+\\.(css|less|scss)$": "identity-obj-proxy",
+            "^.*i18n.*\\.(json)$": "<rootDir>/tests/i18n_mock.json",
+            "^bundle-loader\\?lazy\\!(.*)$": "$1"
+        }
     }
```
Comment (sooraj-shukla) [inline webapp/package.json:63]: Keep the jest config in a separate `jest.config.(js|ts|....)` file instead of adding it in `package.json` file.

**Review by sooraj-shukla — CHANGES_REQUESTED** [review]

File: webapp/src/components/link_tooltip/link_tooltip.jsx (hunk 6 above)
Comment (Nityanand13) [reply webapp/src/components/link_tooltip/link_tooltip.jsx:179]: Discussed

**Review by sooraj-shukla — APPROVED** [review]

File: webapp/src/components/link_tooltip/link_tooltip.jsx [hunk 9]
```diff
@@ -94,94 +97,94 @@ export const LinkTooltip = ({href, connected, gitlabURL}) => {
...
-                                    source={data.description}
-                                    disallowedTypes={['heading']}
-                                    linkTarget='_blank'
-                                />
+                <div className='body d-flex mt-2'>
+                    <span className='pt-1 pb-1 pr-2'>
+                        {getIconElement()}
+                    </span>
+
+                    {/* info */}
```
Comment (avas27JTG) [inline webapp/src/components/link_tooltip/link_tooltip.jsx:125]: remove it or if there is a need to explain anything add descriptive comment

File: webapp/src/components/link_tooltip/link_tooltip.jsx [hunk 10]
```diff
@@ -94,94 +97,94 @@ export const LinkTooltip = ({href, connected, gitlabURL}) => {
...
+                        </a>
+                        <div className='markdown-text mt-1 mb-1'>
+                            <ReactMarkdown
+                                source={data.description}
+                                disallowedTypes={['heading']}
+                                linkTarget='_blank'
+                            />
+                        </div>
+
+                        {/* base <- head */}
```
Comment (avas27JTG) [inline webapp/src/components/link_tooltip/link_tooltip.jsx:139]: same as above

**Review by avas27JTG — APPROVED** [review]

**Review by raghavaggarwal2308 — APPROVED** [review]

**Pull request merged**